algorithms/estonian.sbl

```diff
@@ -0,0 +1,256 @@
+
+externals ( stem )
+booleans ( is_verb )
+integers ( p1 )
+groupings ( V1 RV KI GI)
+stringescapes {}
+
+define V1 'aeiou{o'}{a"}{o"}{u"}'
+define RV 'aeiuo'
+define KI 'kptgbdshfšzž'
+define GI 'cjlmnqrvwxaeiou{o'}{a"}{o"}{u"}'
+define mark_regions as (
+
+    $p1 = limit
+
+    goto V1 gopast non-V1  setmark p1
+)
+
+
+backwardmode (
+
+	define emphasis as (
+	    setlimit tomark p1 for ([substring])
+	    test hop 4 //kingi -> kingi
+            among(
+		'gi' ((GI and not LONGV) delete) //jooksemegi -> jookseme, bioloogi -> bioloogi
+		'ki' (KI delete) //kookki -> kook	   
+	    )
+	    
+	)
+
+	define verb as (
+	    setlimit tomark p1 for ([substring])
+            among(
+		'nuksin' 'nuksime' 'nuksid' 'nuksite' (delete) //seleta-nuksite
+		'ksin' 'ksid' 'ksime' 'ksite' (delete) //personal conditional: rõõmusta-ksin
+		'mata' (delete)
+		'takse' 'dakse' (delete) //impersonal: laul-dakse, luba-takse 
+		'taks' 'daks' (delete) 	//impersonal conditional: laul-daks, saade-taks
+		'akse' (<-'a') 		//impersonal: tulla-kse, süüa-kse, teha-kse, püüt-akse, leita-kse
+		'sime' (delete) 	//pl1pst: saat-sime
+		'site' (delete) 	//pl2pst: saat-site
+		'sin' (delete) 		//sg1pst: laul-sin, saat-sin
+		'me' (V1 delete) 	//pl1prs: laula-me, tule-me
+		'da' (V1 delete) 	//da-infinitive: luba-da
+		'n' (V1 delete) 	//sg1prs: kirjuta-n
+		'b' (V1 delete) 	//sg3prs: laula-b
+	    )
+	    set is_verb
+	)
+
+	define LONGV as 
+		among('aa' 'ee' 'ii' 'oo' 'uu' '{a"}{a"}' '{o"}{o"}' '{u"}{u"}' '{o'}{o'}')
+
+	define i_plural as (
+	    setlimit tomark p1 for ([substring])
+	    among(
+		'i' (RV) //raamatui -> raamatu, lapsikui -> lapsiku
+	    )
+	    delete
+	)
+
+	define special_noun_endings as (
+	    setlimit tomark p1 for ([substring])
+	    among(
+		'lasse' (<- 'lase') //teadlasse -> teadlase
+		'last' (<- 'lase') //teadlast -> teadlase
+		'lane' (<- 'lase') //teadlane -> teadlase
+		'lasi'(<- 'lase') //teadlasi -> teadlase
+		'misse' (<- 'mise') //tegemisse -> tegemise
+		'mist' (<- 'mise') //kasutamist -> kasutamise
+		'mine' (<- 'mise') //tegemine -> tegemise
+		'misi' (<- 'mise') //kasutamisi -> kasutamise
+		'lisse' (<- 'lise') //rohelisse -> rohelise
+		'list' (<- 'lise') //tavalist -> tavalise
+		'line' (<- 'lise') //roheline -> rohelise
+		'lisi' (<- 'lise') //tavalisi -> tavalise
+	     )
+	    
+	)
+	define case_ending as (
+	    setlimit tomark p1 for ([substring])
+            among(
+		'sse' (RV or LONGV) //illative: saapa-sse, tegemisse -> tegemisse
+		'st' (RV or LONGV) //elative: saapa-st, rohelist -> rohelist
+		'le' (RV or LONGV) //allative: raamatu-le
+		'lt' (RV or LONGV) //ablative: raamatu-lt
+		'ga' (RV or LONGV) //komitatiive: õpetaja-ga
+		'ks' (RV or LONGV) //translative: õpetaja-ks
+		'ta' (RV or LONGV) //abessive and da-infinitive: õpetaja-ta and hüpa-ta
+		't' //partitiiv, raamatu-t and kapsas-t
+		's'  (RV or LONGV)//inessive and sg3pst: raamatu-s and sõiti-s
+		'l'  (RV or LONGV) //adessive: raamatu-l and kapsa-l.
```

Comment:
Are you aware the action applied here in **every** matching case will be the same?  I could see the logic in putting the action just once or putting it explicitly for every case, but putting it against all but one case makes me wonder if you've misunderstood this feature of `among` and are expecting the above to mean no action for `t` or something.

Reply:
Reading the description of the algorithm in the PR for the website, it seems that `t` is meant to be different, and the example of `kapsas-t` seems to confirm that.  However that example will actually be handled as `kapsa-st` (since the longest suffix which matches is chosen).

Assuming you want `t` not to use the code following `s` then you can add `()` after `t` so it uses that empty code instead.  If it should have the `(RV or LONGV)` check then the description for the website needs adjusting.

Also, is `sg3pst` in the comment after `s` a typo?

Reply:
> Also, is `sg3pst` in the comment after `s` a typo?

Ah no, probably shorthand for "singular third-person past tense".